nvflare/fuel/f3/cellnet/core_cell.py

```diff
@@ -1124,15 +1124,16 @@ def _send_to_endpoint(self, to_endpoint: Endpoint, message: Message) -> str:
                 err = ReturnCode.MSG_TOO_BIG
             else:
                 direct_cell = self.ALL_CELLS.get(to_endpoint.name)
-                msg_size_mbs = self._msg_size_mbs(message)
                 if direct_cell:
                     # create a thread and fire the cell's process_message!
                     # self.DIRECT_MSG_EXECUTOR.submit(self._send_direct_message, direct_cell, message)
                     self._send_direct_message(direct_cell, message)
 
                 else:
                     self.communicator.send(to_endpoint, CoreCell.APP_ID, message)
-                self.sent_msg_size_pool.record_value(category=self._stats_category(message), value=msg_size_mbs)
+                self.sent_msg_size_pool.record_value(
+                    category=self._stats_category(message), value=self._msg_size_mbs(message)
+                )
```

Comment:
need to keep the original change in PR https://github.com/NVIDIA/NVFlare/pull/2093
which means calculate the msg size BEFORE "_send_direct_message"

Reply:
I see. Will fix.

Reply:
Fixed